src/audio/CMakeLists.txt

```diff
@@ -174,19 +174,20 @@ set(sof_audio_modules mixer volume src asrc eq-fir eq-iir dcblock crossover tdfb
 # sources for each module
 if(CONFIG_IPC_MAJOR_3)
 	set(volume_sources volume/volume.c volume/volume_generic.c volume/volume_ipc3.c)
+	set(asrc_sources asrc/asrc.c asrc/asrc_farrow.c asrc/asrc_farrow_generic.c asrc/asrc_ipc3.c)
 	set(src_sources src/src.c src/src_ipc3.c src/src_generic.c)
 	set(eq-iir_sources eq_iir/eq_iir_ipc3.c eq_iir/eq_iir_generic.c)
 	set(eq-fir_sources eq_fir/eq_fir_ipc3.c)
 	set(tdfb_sources tdfb/tdfb_ipc3.c)
 elseif(CONFIG_IPC_MAJOR_4)
 	set(volume_sources volume/volume.c volume/volume_generic.c volume/volume_ipc4.c)
+	set(asrc_sources asrc/asrc.c asrc/asrc_farrow.c asrc/asrc_farrow_generic.c asrc/asrc_ipc4.c)
```

Comment:
please check eq-fir_sources for example:  share the common "set(asrc_sources asrc/asrc.c asrc/asrc_farrow.c asrc/asrc_farrow_generic.c)" after this version check , only set "asrc/asrc_ipc3.c or asrc/asrc_ipc4.c" here,